armada_jupyter/submissions.py

```diff
@@ -0,0 +1,205 @@
+    Manages the resources for a Kubernetes pod ran by Armada
+    """
+
+    def __init__(
+        self, limits: K8sResourceOptions = None, requests: K8sResourceOptions = None
+    ):
+        if limits is None and requests is None:
+            raise ValueError("Must specify at least one of limits or requests")
+
+        self.limits = limits
+        self.requests = limits
+
+    def __repr__(self) -> str:
+        return f"K8sResources({YMLSTR.limits}={self.limits}, {YMLSTR.requests}={self.requests})"
+
+
+class Submission:
+    """
+    Represents a Armada-Jupyter Submission
+    """
+
+    def __init__(
+        self,
+        name: str,
+        image: str,
+        armada_queue: str,
+        armada_priority: int,
+        timeout: str,
+        resources: Optional[K8sResources] = None,
+    ):
+        self.name = name
+        self.image = image
+        self.armada_queue = armada_queue
+        self.armada_priority = armada_priority
+        self.timeout = self.timeout_conversion(timeout)
+        self.resources = resources
+
+        # Check that resources is the right object
+        if not isinstance(resources, K8sResources) and resources is not None:
+            raise ValueError(
+                f"resources must be a K8sResources object, not {type(resources)}"
+            )
+
+    @staticmethod
+    def timeout_conversion(timeout: str) -> Optional[str]:
+        """
+        Convert timeout from minutes and hours into seconds.
+
+        timeout can be in the form of 10m or 5h for example
+
+        should return a string like 100s
+        """
+
+        if timeout[-1] == "m":
+            return str(int(timeout[:-1]) * 60) + "s"
+        elif timeout[-1] == "h":
+            return str(int(timeout[:-1]) * 60 * 60) + "s"
+        elif timeout[-1] == "s":
+            return timeout
+
+        ValueError("Timeout must be in the form of [x]h, [x]m or [x]s")
+        return None
+
+    def resources_from_dict(
+        self, resources: Dict[str, Dict[str, Union[str, int]]]
+    ) -> None:
+        """
+        Used for adding a resources from dict
+        """
+
+        self.resources = k8s_resources_from_dict(
+            limits=resources.get(YMLSTR.limits),
+            requests=resources.get(YMLSTR.requests),
+        )
+
+    def __repr__(self) -> str:
+        return (
+            f"Submission(f{YMLSTR.name}='{self.name}', {YMLSTR.image}='{self.image}' "
+            f"{YMLSTR.armada_queue}='{self.armada_queue}', "
+            f"{YMLSTR.armada_priority}={self.armada_priority}, "
+            f"{YMLSTR.timeout}='{self.timeout}', {YMLSTR.resources}={self.resources})"
+        )
+
+
+def k8s_resources_from_dict(
+    limits: Optional[Dict[str, Union[str, int]]] = None,
+    requests: Optional[Dict[str, Union[str, int]]] = None,
+) -> K8sResources:
+    """
+    Creates a K8sResources object from a dict
+    """
+
+    changed: Dict[str, Optional[K8sResourceOptions]] = {}
+
+    if limits is None and requests is None:
```

Comment:
We should check but I've never really limits and requests sepately.  They always come together in Armada examples.

Reply:
Okay, I will change that to match. From looking at the k8s spec more closely I think this is the case.